drivers/clock_control/stm32f0x_ll_clock.c

```diff
@@ -0,0 +1,81 @@
+/*
+ *
+ * Copyright (c) 2017 Linaro Limited.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <soc.h>
+#include <soc_registers.h>
+#include <clock_control.h>
+#include <misc/util.h>
+#include <clock_control/stm32_clock_control.h>
+#include "stm32_ll_clock.h"
+
+
+#ifdef CONFIG_CLOCK_STM32_SYSCLK_SRC_PLL
+
+/**
+ * @brief fill in pll configuration structure
+ */
+void config_pll_init(LL_UTILS_PLLInitTypeDef *pllinit)
+{
+	/*
+	 * PLL MUL
+	 * 2  -> LL_RCC_PLL_MUL_2  -> 0x00000000
+	 * 3  -> LL_RCC_PLL_MUL_3  -> 0x00040000
+	 * 4  -> LL_RCC_PLL_MUL_4  -> 0x00080000
+	 * ...
+	 * 16 -> LL_RCC_PLL_MUL_16 -> 0x00380000
+	 */
+	pllinit->PLLMul = ((CONFIG_CLOCK_STM32_PLL_MULTIPLIER - 2)
+					<< RCC_CFGR_PLLMUL_Pos);
+#if defined(RCC_PLLSRC_PREDIV1_SUPPORT)
+	/* PREDIV support is a specific RCC configuration present on */
+	/* following SoCs: STM32F070x6, STM32F070xB and STM32F030xC */
+	/* cf Reference manual for more details */
+#if defined(CONFIG_CLOCK_STM32_PLL_SRC_HSI)
+	pllinit->PLLDiv = LL_RCC_PLLSOURCE_HSI_DIV_2;
+#else
+	/*
+	 * PLL DIV
+	 * 1  -> LL_RCC_PLLSOURCE_HSE_DIV_1  -> 0x00010000
+	 * 2  -> LL_RCC_PLLSOURCE_HSE_DIV_2  -> 0x00010001
+	 * 3  -> LL_RCC_PLLSOURCE_HSE_DIV_3  -> 0x00010002
+	 * ...
+	 * 16 -> LL_RCC_PLLSOURCE_HSE_DIV_16 -> 0x0001000F
+	 */
+	pllinit->PLLDiv = (RCC_CFGR_PLLSRC_HSE_PREDIV |
+					(CONFIG_CLOCK_STM32_PLL_PREDIV1 - 1));
```

Comment:
CLOCK_STM32_PLL_PREDIV1 is not available in Kconfig.
If it should be (as it seems), maybe  we can just re-use STM32F3 Kconfig options
...and maybe this whole file (if same exact same code is used)

Reply:
This is added in the stm32f091 support from @b0661, I will pick his change for that.

Reply:
Done, I merged it in the clock support patch

Reply:
Are there good reasons not to use stm32f3 file, as this is exact same code?
Do you foresee divergences in the future?

Reply:
I dont't have any opinion, should I update it ? the PR is ok to be merged, maybe it's something that can be cleaned up later on ?

Reply:
ok, let's keep it that way for now. With time, we'll know if this is a good thing to do.